libs/play/index.js

```diff
@@ -0,0 +1,215 @@
+import * as express from "express";
+import * as crypto from "node:crypto";
+
+import he from "he";
+
+export const ORIGIN_PLAY = process.env["ORIGIN_PLAY"] || "localhost";
+export const ORIGIN_MAIN = process.env["ORIGIN_MAIN"] || "localhost";
+
+/** @import { IncomingMessage, ServerResponse } from "http" */
+
+/**
+ * @typedef State
+ * @property {string} html
+ * @property {string} css
+ * @property {string} js
+ * @property {string} [src]
+ */
+
+/**
+ * @param {IncomingMessage | null} _proxyRes
+ * @param {IncomingMessage} _req
+ * @param {ServerResponse<IncomingMessage>} res
+ */
+export function withRunnerResponseHeaders(_proxyRes, _req, res) {
```

Comment:
As it's only used below, we could change this to something like:

```suggestion
function setRunnerResponseHeaders(res) {
```